pipelines/curate_identity.py

```
@@ -560,3 +985,353 @@ def curate_records(records: Iterable[SourceRecord]) -> tuple[CurationResult, ...
+    descriptor = os.open(path, os.O_WRONLY | os.O_CREAT | os.O_EXCL, 0o644)
+    try:
+        with os.fdopen(descriptor, "wb") as handle:
+            handle.write(payload)
+            handle.flush()
+            os.fsync(handle.fileno())
+    except BaseException:
+        path.unlink(missing_ok=True)
+        raise
+
+
+def validate_identity_tree(dest: Path) -> FactoryRegistry:
```

Ruff (0.16.2)

[warning] Too many branches (21 > 12)

(PLR0912)